[quant][graphmode][fx] Merge quant_env and env

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59042 [quant][graphmode][fx][refactor] Remove _convert from Quantizer class
* #59041 [quant][graphmode][fx][fix] Fix support for custom module
* #59040 [quant][graphmode][fx][refactor] Remove Quantizer class from convert (QuantizeHandler)
* #59039 [quant][graphmode][fx][refactor] Remove Quantizer class from quantize_node
* #59038 [quant][graphmode][fx][refactor] Remove _prepare from Quantizer class
* #59037 [quant][graphmode][fx][refactor] Remove find_matches from Quantizer class
* #59036 [quant][graphmode][fx][refactor] Remove fold_weights from Quantizer class
* #59035 [quant][graphmode][fx][refactor] Remove save_state and restore_state from Quantizer class
* #59034 [quant][graphmode][fx][refactor] Remove prepare_custom_config from Quantizer class
* #59033 [quant][graphmode][fx][refactor] Remove patterns from Quantizer class
* #59032 [quant][graphmode][fx][refactor] Remove node_name_to_scope from Quantizer
* #59031 [quant][graphmode][fx][refactor] Remove quantized_graph from Quantizer
* #59030 [quant][gx][graphmode][refactor] Remove modules from Quantizer
* #59029 [quant][graphmode][fx] Factor out run_weight_observer
* **#59028 [quant][graphmode][fx] Merge quant_env and env**

Summary:
Previously we have an env and a quant_env in convert, which is a bit confusing,
in this PR we merged them and have a Dict[str, Tuple[Node, torch.dtype]]

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724863](https://our.internmc.facebook.com/intern/diff/D28724863)

jerryzh168 committed:
[quant][graphmode][fx] Merge quant_env and env

Summary:
Previously we have an env and a quant_env in convert, which is a bit confusing,
in this PR we merged them and have a Dict[str, Tuple[Node, torch.dtype]]

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724863).

vkuzo commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -1083,8 +1084,10 @@ def should_insert_observer_for_output(
     ) -> bool:
-        return True
+        # FixQParamOps are the same as CopyNode in int8 quantization
+        return activation_dtype(qconfig) in [torch.quint8, torch.qint8]
```
is this a behavior change? or is the end result the same and this was done elsewhere before?

jerryzh168 replied:
Yes, the end result is the same I think (no changes to test), but previous code probably works by coincidence I guess, since FixedQParams can't be treated as CopyNode when we do fp16 quantization.

vkuzo reviewed (approved):
accepting to unblock!

jerryzh168 committed:
Update on "[quant][graphmode][fx] Merge quant_env and env"


Summary:
Previously we have an env and a quant_env in convert, which is a bit confusing,
in this PR we merged them and have a Dict[str, Tuple[Node, torch.dtype]]

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724863](https://our.internmc.facebook.com/intern/diff/D28724863)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724863).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Merge quant_env and env"


Summary:
Previously we have an env and a quant_env in convert, which is a bit confusing,
in this PR we merged them and have a Dict[str, Tuple[Node, torch.dtype]]

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724863](https://our.internmc.facebook.com/intern/diff/D28724863)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724863).

jerryzh168 committed:
Update on "[quant][graphmode][fx] Merge quant_env and env"


Summary:
Previously we have an env and a quant_env in convert, which is a bit confusing,
in this PR we merged them and have a Dict[str, Tuple[Node, torch.dtype]]

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28724863](https://our.internmc.facebook.com/intern/diff/D28724863)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28724863).
